doc/src/content/xdocs/spec.xml

```diff
@@ -328,15 +328,19 @@
     <section>
       <title>Data Serialization</title>
 
-      <p>Avro data is always serialized with its schema.  Files that
-	store Avro data should always also include the schema for that
-	data in the same file.  Avro-based remote procedure call (RPC)
-	systems must also guarantee that remote recipients of data
-	have a copy of the schema used to write that data.</p>
-
-      <p>Because the schema used to write data is always available
-	when the data is read, Avro data itself is not tagged with
-	type information.  The schema is required to parse data.</p>
+      <p>Avro data does not include type information.
+  The binary encoding does not include field names, either. As a
+  result, the original schema of the writer is always required in
+  order to read Avro data accurately.</p>
+
+      <p>Therefore, files that store Avro data should always include the
+  schema for that data in the same file.  Avro-based remote procedure
+  call (RPC) systems must also guarantee that remote recipients of data
+	have a copy of the schema used to write that data.  In general, any
+  reader of Avro data must have the same schema (as defined more fully
```

Comment:
I think it would be more correct to say must have a "structurally identical" schema. Parsing canonical form defines how to produce a string you can hash for a reliable fingerprint, but you can still read data as long as the type structure is the same. For example, I can read a record Point with two long fields, x and y, using the record schema Pair with long fields a and b. This is common for reading a file of data written with generic using a reflected class.